src/lando/api/legacy/workers/uplift_worker.py

```diff
@@ -0,0 +1,235 @@
+    """
+
+    job_type = UpliftJob
+
+    worker_type = WorkerType.UPLIFT
+
+    @override
+    def refresh_active_repos(self):
+        """Override base functionality by not checking treestatus."""
+        self.active_repos = self.enabled_repos
+
+    @override
+    def run_job(self, job: UpliftJob) -> bool:
+        """Run an uplift job."""
+        repo = job.target_repo
+        user = job.multi_request.user
+        job_url = job.url()
+
+        try:
+            created_revision_ids = self.apply_and_uplift(job)
+        except TemporaryFailureException:
+            return False
+        except PermanentFailureException as exc:
+            self.notify_uplift_failure(job, repo.name, job_url, user.email, str(exc))
+            return False
+        except Exception as exc:  # pragma: no cover - defensive catch
+            self.notify_uplift_failure(job, repo.name, job_url, user.email, str(exc))
+            return False
+
+        self.notify_success(
+            repo.name,
+            job_url,
+            user.email,
+            created_revision_ids,
+        )
+        return True
+
+    def apply_and_uplift(self, job: UpliftJob) -> list[int]:
+        """Apply uplift patches to the repo and create new revisions.
+
+        Returns an ordered list of created revision IDs.
+        """
+        repo = job.target_repo
+        multi_request = job.multi_request
+        user = multi_request.user
+        scm = repo.scm
+
+        # Update to the latest commit in the target train.
+        base_revision = self.update_repo(repo, job, scm, target_cset=None)
+
+        # Apply patches to the tip of the target train.
+        def apply_patch(revision: Revision):
+            logger.debug(f"Landing {revision} ...")
+            scm.apply_patch(
+                revision.diff,
+                revision.commit_message,
+                revision.author,
+                revision.timestamp,
+            )
+
+        for uplift_revision in job.revisions.all():
+            self.handle_new_commit_failures(
+                apply_patch, repo, job, scm, uplift_revision
+            )
+            new_commit = scm.describe_commit()
+            logger.debug(f"Created new commit {new_commit}")
+
+        # On success: create patches.
+        response = self.moz_phab_uplift(
+            job, user.profile.phabricator_api_key, base_revision
+        )
+
+        # Retrieve created revision IDs and tip revision ID.
+        commits = response["commits"]
+        created_revision_ids = [int(commit["rev_id"]) for commit in commits]
+        tip_revision_id = created_revision_ids[-1]
+
+        _, created = UpliftRevision.objects.get_or_create(
+            revision_id=tip_revision_id,
+            defaults={"assessment": multi_request.assessment},
+        )
+        if not created:
+            UpliftRevision.objects.filter(revision_id=tip_revision_id).update(
+                assessment=multi_request.assessment
+            )
```

Comment:
What's the point of using `get_or_create` here, if we are ignoring the fetched object?
```suggestion
        uplift_revision, created = UpliftRevision.objects.get_or_create(
            revision_id=tip_revision_id,
            defaults={"assessment": multi_request.assessment},
        )
        if not created:
            uplift_revision.assessment=multi_request.assessment
            uplift_revision.save()
```

Reply:
Switched back to using a regular `create`. I added this to allow jobs to be re-run because I noticed a failure in the tests. However that failure was happening because we mock out `moz-phab uplift` and it was returning the same revision numbers twice, creating a conflict when trying to add another `UpliftRevision` with the same `revision_id`. In reality `moz-phab uplift` will always return new revision numbers, so we can just do `create` instead. I updated the test to check more correct conditions as well.